### PR TITLE
Added feature to save estimates and trajectory visualization

### PR DESCRIPTION
**Pull request by MichaelMarav**
### Description
* Added option to save the serow base state estimates (position + orientation) into a txt file. 
* Added README that gives instructions on how to visualize the test results using evo package
* Added folder test/results for saving the base estimates
* Corrected the mass of go2 on .json by adding the base mass that pinocchio ignores.
* Tested visually if the mass of the robot affects the estimation and it does not (8kg vs 15kg).

### CODE REVIEW

File: test/go2_test.cpp [hunk 1]
```diff
@@ -20,6 +20,9 @@
 #include <sstream>
 #include <string>
 
+const bool STORE_PREDICTIONS = true; // If true the serow base estimates are stored under results/
```
Comment (mrsp) [inline test/go2_test.cpp:23]:
```suggestion
constexpr bool kStorePredictions = true; // If true the serow base estimates are stored under results/
```
https://google.github.io/styleguide/cppguide.html#Constant_Names

File: test/go2_test.cpp [hunk 2]
```diff
@@ -162,5 +177,27 @@ TEST(SerowTests, Go2Test) {
                   << std::endl;
         std::cout << "CoM external forces " << state->getCoMExternalForces().transpose()
                   << std::endl;
+
+        // Store the results into an eigen matrix to be saved
+        if (STORE_PREDICTIONS) 
+        {
+            if (timestamp - prev_timestamp < 0){
+                sec_timer += 1.;
```
Comment (mrsp) [inline test/go2_test.cpp:185]: +1?

Comment (MichaelMarav) [reply test/go2_test.cpp:185]: Yes, if time has moved backwards 0.998, 0.999 , 0.001 then add one second to the sec_timer else you are in the same second duration (e.g. third second 3 + 0.580, 3 + 0.581, 3 + 0.582 .... 3 + 0.999, 4 + 0.001). This is for plotting in continuous one-directional timeline

File: test/README.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
+## Trajectory Visualization Overview
+When running a script at test/build for a robot dataset, the serow base state estimates (position + orientation) are saved at *results/<name>.txt* if the flag STORE_PREDICTIONS is true. In order to visualize the predictions follow the instructions below
```
Comment (mrsp) [inline test/README.md:2]:
```suggestion
When running a script at test/build for a robot dataset, the serow base state estimates (position + orientation) are saved at *results/<name>.txt* if the flag kStorePredictions is true. In order to visualize the predictions follow the instructions below
```

File: test/go2_test.cpp [hunk 4]
```diff
@@ -28,16 +31,26 @@ TEST(SerowTests, Go2Test) {
...
-    std::fstream file("../../test/data/go2.csv", std::ios::in);
-    if (!file.is_open()) {
+    std::fstream inputFile("../data/go2.csv", std::ios::in); // Input file
+    std::ofstream outputFile;
+    if (!inputFile.is_open()) {
         throw std::runtime_error("Could not open file");
         return;
     }
+
+    if (STORE_PREDICTIONS){
```
Comment (mrsp) [inline test/go2_test.cpp:41]:
```suggestion
    if (kStorePredictions){
```

File: test/go2_test.cpp [hunk 5]
```diff
@@ -162,5 +177,27 @@ TEST(SerowTests, Go2Test) {
                   << std::endl;
         std::cout << "CoM external forces " << state->getCoMExternalForces().transpose()
                   << std::endl;
+
+        // Store the results into an eigen matrix to be saved
+        if (STORE_PREDICTIONS)
```
Comment (mrsp) [inline test/go2_test.cpp:182]:
```suggestion
        if (kStorePredictions) 
```

**Pull request merged**